### ISSUE TITLE
Loot Table Manager error with Buried Treasure

### ISSUE THREAD

**Issue by katubug** [Phylogeny/BoundTotems#4]
Forge Version: 31.2.31
Bound Totems: 1.3.0

Hi! I'm trying to diagnose an issue where loot fails to generate in a handful of chests, sporadically. While I do not think it is your mod causing it, I did come across this in my server log, and thought I should report:

```
[08Jul2020 20:07:39.764] [Server thread/ERROR] [net.minecraft.world.storage.loot.LootTableManager/]: Couldn't parse loot table minecraft:chests/buried_treasure
java.lang.NullPointerException: null
	at net.minecraft.world.storage.loot.LootTable.lambda$addPool$2(LootTable.java:236) ~[?:?]
	at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90) ~[?:1.8.0_242]
	at java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1359) ~[?:1.8.0_242]
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126) ~[?:1.8.0_242]
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499) ~[?:1.8.0_242]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486) ~[?:1.8.0_242]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_242]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230) ~[?:1.8.0_242]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196) ~[?:1.8.0_242]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_242]
	at java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:516) ~[?:1.8.0_242]
	at net.minecraft.world.storage.loot.LootTable.addPool(LootTable.java:236) ~[?:?]
	at com.github.phylogeny.boundtotems.CommonEvents.modifyChestLootTables(CommonEvents.java:47) ~[?:1.15.2-1.3.0]
	at net.minecraftforge.eventbus.ASMEventHandler_361_CommonEvents_modifyChestLootTables_LootTableLoadEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:80) ~[eventbus-2.2.0-service.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:258) ~[eventbus-2.2.0-service.jar:?]
	at net.minecraftforge.event.ForgeEventFactory.loadLootTable(ForgeEventFactory.java:642) ~[?:?]
	at net.minecraftforge.common.ForgeHooks.loadLootTable(ForgeHooks.java:834) ~[?:?]
	at net.minecraft.world.storage.loot.LootTableManager.lambda$apply$0(LootTableManager.java:46) ~[?:?]
	at java.util.HashMap.forEach(HashMap.java:1289) ~[?:1.8.0_242]
	at net.minecraft.world.storage.loot.LootTableManager.func_212853_a_(LootTableManager.java:44) ~[?:?]
	at net.minecraft.world.storage.loot.LootTableManager.func_212853_a_(LootTableManager.java:22) ~[?:?]
	at net.minecraft.client.resources.ReloadListener.func_215269_a(SourceFile:13) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:670) ~[?:1.8.0_242]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:646) ~[?:1.8.0_242]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456) ~[?:1.8.0_242]
	at net.minecraft.resources.AsyncReloader.func_219557_a(SourceFile:71) ~[?:?]
	at net.minecraft.util.concurrent.TickDelayedTask.run(SourceFile:18) [?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) [?:?]
	at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23) [?:?]
	at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:731) [?:?]
	at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:141) [?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) [?:?]
	at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:714) [?:?]
	at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:708) [?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127) [?:?]
	at net.minecraft.server.MinecraftServer.func_195568_a(MinecraftServer.java:1464) [?:?]
	at net.minecraft.server.MinecraftServer.func_195560_a(MinecraftServer.java:450) [?:?]
	at net.minecraft.server.MinecraftServer.func_71247_a(MinecraftServer.java:363) [?:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:212) [?:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:613) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
```

That is the only relevant portion, I think. But the full log is available here: https://paste.ee/p/JeKpq

**Comment by graeye** [comment]
This may be caused by unnamed loot pools, which I've confirmed are present in this mod. We are running into this breaking the majority of vanilla chests (always 100% empty chests), confirmed resolved by removing this mod from the pack.

Mod author, please see this thread to see if it's relevant (post by SilentChaos512). He discusses how to fix it in code. https://www.curseforge.com/minecraft/mc-mods/explorercraft-worldexpansion?comment=123

tl;dr

> SilentChaos512
My guess (since I can't see your code...) is that you are adding loot to some loot tables, but aren't putting a name on your loot pools. You should call `name` on `LootPool.Builder` and give the pools a unique name, [like this](https://github.com/SilentChaos512/SilentGems/blob/1.14/src/main/java/net/silentchaos512/gems/init/GemsLoot.java#L69).

**Comment by Phylogeny** [comment]
Thank you @katubug and @graeye for troubleshooting and figuring out how to fix this bug.
